x/auth/vesting/types/vesting_account_test.go

```diff
@@ -924,3 +925,585 @@ func initBaseAccount() (*authtypes.BaseAccount, sdk.Coins) {
+		},
+		{
+			name:             "update at exactly the vesting end time",
+			startTime:        now.Add(-24 * time.Hour).Unix(),
+			endTime:          now.Unix(), // End time is now
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 150)), // Original 100 + 50 new
+			expectedEndTime:  now.Unix(),
+			testTime:         now.Unix(),
+		},
+		{
+			name:             "multiple denominations in original vesting",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 150)),
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "zero rewards",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 0)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)), // No change
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "nil rewards",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      nil,
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)), // No change
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "start time equals end time (zero duration)",
+			startTime:        now.Unix(),
+			endTime:          now.Unix(), // Zero duration
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 150)), // Should still update vesting amount
```

Comment:
why? If the start time == end time then the vesting is complete and I would expect expectedVesting to be 100 (the original)